cmd/esc/cli/env_clone.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023, Pulumi Corporation.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/esc/cmd/esc/cli/client"
+	"github.com/spf13/cobra"
+)
+
+func newEnvCloneCmd(env *envCommand) *cobra.Command {
+	var (
+		preserveHistory         bool
+		preserveAccess          bool
+		preserveEnvironmentTags bool
+		preserveRevisionTags    bool
+	)
+
+	cmd := &cobra.Command{
+		Use:   "clone [<org-name>/]<project-name>/<environment-name> [<project-name>/]<environment-name>",
+		Args:  cobra.MaximumNArgs(2),
+		Short: "Clone an existing environment into a new environment.",
+		Long: "Clone an existing environment into a new environment.\n" +
+			"\n" +
+			"This command clones an existing environment with the given identifier into a new environment.\n" +
+			"If a project is omitted from the new environment identifier the new environment will be created\n" +
+			"within the same project as the environment being cloned.\n",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := env.esc.getCachedClient(ctx); err != nil {
+				return err
+			}
+
+			ref, args, err := env.getNewEnvRef(ctx, args)
```

Comment:
this is the source ref right? this should be `getExistingEnvRef`

Reply:
now that i think about it, do we even want the legacy path defaulting (org/env)? If not, we might just want to call `getEnvRef` or `parseRef` directly

Reply:
ah yeah good catch - I had to add a new field onto the environmentRef struct because I want to be able to explicitly fail if users are passing in just the environment name (omitting the project)

Reply:
it's not pretty but I did a similar check like [this](https://github.com/pulumi/esc/blob/d3ce3545515bffea9656d58f4ae7bc0849bc3d40/cmd/esc/cli/env.go#L173) without using a field, would that work here? otherwise what you have here is also fine

Reply:
Hmm I could but it still wouldn't tell me if it's ambiguous because of the org or the project name since we would set the project to be the default value and it wouldn't be clear if we set it or the user did